src/SpendPermissionManager.sol

```diff
@@ -95,6 +96,10 @@ contract SpendPermissionManager is EIP712 {
     /// @notice ERC-7528 address convention for native token (https://eips.ethereum.org/EIPS/eip-7528).
     address public constant NATIVE_TOKEN = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
 
+    /// @notice A flag to indicate if the contract can receive native token transfers.
+    /// @dev Contract can only receive during the execution of `spend` for native tokens.
+    bool canReceive = false;
```

Comment:
Would also appreciate a pas for gas optimizing this case. Transient storage is designed exactly for a case like this so I would opt to use it! Seems like more recent solidity versions support `transient` keyword for base types so we may be in luck for an easy devx :)
https://docs.soliditylang.org/en/latest/contracts.html#transient-storage

Reply:
as discussed also on slack, I'm going to abort this attempt for now because the fixed dependency (MagicSpend) at =0.8.23 is incompatible with the versions we would need to get access to either transient storage opcodes (`tstore`, `tload` etc)  (minimum 0.8.24) or the `transient` keyword (min 0.8.28)